pkg/image/external/external.go

```diff
@@ -0,0 +1,160 @@
+package external
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sort"
+	"strings"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/rancher/rancher/pkg/controllers/management/k3sbasedupgrade"
+	"github.com/sirupsen/logrus"
+)
+
+type Source string
+
+const (
+	K3S  Source = "k3s"
+	RKE2 Source = "rke2"
+)
+
+var (
+	RKE2MinimumKubernetesVersion = &semver.Version{
+		Major: 1,
+		Minor: 21,
+		Patch: 0,
+	}
+)
+
+func GetExternalImages(rancherVersion string, externalData map[string]interface{}, source Source, os, arch string) ([]string, error) {
+	if source != K3S && source != RKE2 {
+		return nil, fmt.Errorf("invalid source provided: %s", source)
+	}
+
+	logrus.Infof("generating %s image list...", source)
+	externalImagesMap := make(map[string]bool)
+	releases, _ := externalData["releases"].([]interface{})
+	var compatibleReleases []string
+
+	for _, release := range releases {
+		releaseMap, _ := release.(map[string]interface{})
+		version, _ := releaseMap["version"].(string)
+		if version == "" {
+			continue
+		}
+
+		// Check that RKE2 uses at least Kubernetes v1.21+.
+		if source == RKE2 {
+			versionSemVer, err := semver.NewVersion(strings.TrimPrefix(version, "v"))
+			if err != nil {
+				continue
+			}
+			if versionSemVer.LessThan(*RKE2MinimumKubernetesVersion) {
+				continue
+			}
+		}
```

Comment:
I think you mentioned it before but why are we checking that the version is > 1.21?

Reply:
RKE2 Provisioning will only be supported on Kubernetes v1.21+. The `rke2-images-all` file is only available on v1.21+ as well: https://github.com/rancher/rke2/releases/

Reply:
Can you put a comment above where the variable is declared explaining this?

Reply:
Yes! Saw this recently, and will do.
https://github.com/rancher/rancher/pull/32916#pullrequestreview-670860012